### PR TITLE
Created loading component and related file

### LINKED ISSUE #31: Loading Component
**Describe the task that needs to be done.**
Adding a loading component for when a screen is rendering.

**Describe how a solution to your proposed task might look like (and any alternatives considered).**
Expect the loading component to be added as an overlay to the main page component, and have the visibility toggled using a React state or something that will be set alongside all needed API calls that we are making.

**Notes**
Feel free to message me if there's any thoughts about how this might proceed :)

### PR DESCRIPTION
**Pull request by uoa-lzha989**
Addition of loading component and the unit test

## Description
Fixes issue no. 31, creating the loading component. Will be able to be used as an overlay component for when loading is occurring. May require some changes once actually being implemented in the future, as it is currently difficult to test with no additional setup (currently sets up the basics).

Fixes #31 

## How has this been tested?
Unit test has been written to verify it loading.

## Screenshots
image

## Checklist
*(Leave blank if not applicable)*

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules
- [ ] I have checked my code and corrected any misspellings

### CODE REVIEW

File: src/components/Loading/LoadingComponent.jsx [hunk 1]
```diff
@@ -0,0 +1,17 @@
+import React from 'react'
+import CircularProgress from '@mui/material/CircularProgress';
+
+function LoadingComponent() {
+  return (
+    <div style={{
```
Comment (scorpionknifes) [inline src/components/Loading/LoadingComponent.jsx:6]: convert inline css to `LoadingComponent.module.css`

feel free to look at other components for consistency

File: src/components/Loading/LoadingComponent.jsx [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+        display: 'flex',
+        alignItems: 'center',
+        justifyContent: 'center',
+        height: '50vh'
```
[the remark this answers is not in the document]
Comment (uoa-lzha989) [reply src/components/Loading/LoadingComponent.jsx:10]: I did try to use 100% but the placement within the screen right now isn't ideal - it doesn't actually flex it.

**Comment by uoa-lzha989** [comment]
RE: CZ's comment about 100% vs 50vh - I won't change the value yet as that change doesn't really do anything yet - once the component is actually being integrated further, if we run into some funny business, I'll then work with whoever (if it isn't me) to integrate a new change should we need it - but visually not very viable rn IMO.

**Review by scorpionknifes — APPROVED** [review]

**Review by DanilLinkov — APPROVED** [review]
LGTM

**Pull request merged**